### ISSUE TITLE
IOS "Home couldn't connect to this accessory"

### ISSUE THREAD

**Issue by talisker-ca** [KraigM/homebridge-nest#94]
I posted this on the homebridge page originally #947
... and was advised by KhaosT ...
_"This is normally caused by plugin returns invalid values for accessory. Please contact developer of homebridge-nest to sort this out."_

so I am reposting here ...

I decided to try to make use of HomeBridge to 'initially' connect to my Nest devices in the house.
I installed HomeBridge on my Linux server (OEL7) which had some of its own challenges, but eventually installed ok.  I added the Nest plugin, and configured it according to the instructions ... 

config.json file...
{
        "bridge": {
                "name": "Homebridge",
                "username": "CC:22:3D:E3:XX:XX",
                "port": 38XXX,
                "pin": "031-XX-XXX"
        },

        "description": "This is the configuration file for my devices",

        "platforms": [{
                "platform": "Nest",
                "token": "c.h4AEXXX..."
        }]
}

It certainly _seems_ to connect to the Nest device(s) ok ... 

# homebridge 
[11/23/2016, 6:34:07 PM] Loaded plugin: homebridge-nest
[11/23/2016, 6:34:07 PM] Registering platform 'homebridge-nest.Nest'
[11/23/2016, 6:34:07 PM] ---
[11/23/2016, 6:34:07 PM] Loaded config.json with 0 accessories and 1 platforms.
[11/23/2016, 6:34:07 PM] ---
[11/23/2016, 6:34:07 PM] Loading 1 platforms...
[11/23/2016, 6:34:07 PM] Initializing Nest platform...
[11/23/2016, 6:34:07 PM] Fetching Nest devices.
[11/23/2016, 6:34:08 PM] Software version for Family Room Thermostat (Nest) is: 5.6-7
[11/23/2016, 6:34:08 PM] Current temperature for Family Room Thermostat (Nest) is: 20.5 C
[11/23/2016, 6:34:08 PM] Current humidity for Family Room Thermostat (Nest) is: 30%
[11/23/2016, 6:34:08 PM] Target temperature for Family Room Thermostat (Nest) is: 21 C
[11/23/2016, 6:34:08 PM] Target heating for Family Room Thermostat (Nest) is: Heating
[11/23/2016, 6:34:08 PM] Away for Family Room Thermostat (Nest) is: 0
[11/23/2016, 6:34:08 PM] Software version for Hallway Nest Protect is: 3.1rc9
[11/23/2016, 6:34:08 PM] Software version for Basement Nest Protect is: 3.1rc9
[11/23/2016, 6:34:08 PM] Initializing platform accessory 'Family Room Thermostat (Nest)'...
[11/23/2016, 6:34:08 PM] Initializing platform accessory 'Hallway Nest Protect'...
[11/23/2016, 6:34:08 PM] Initializing platform accessory 'Basement Nest Protect'...
Scan this code with your HomeKit App on your iOS device to pair with Homebridge:
031-XX-XXX
[11/23/2016, 6:34:09 PM] Homebridge is running on port 38XXX.
[11/23/2016, 7:01:48 PM] Current temperature for Family Room Thermostat (Nest) is: 21 C

but the Home app on IOS does not want to connect to HomeBridge ... 

the Accessory 'Homebridge" shows up ok, when you click it you get the warning message "This accessory is not certified and may not work reliably with HomeKit"   selected "add anyway" ... and manually entered the 8 digit code  which eventually results in the message "Couldn't add Homebridge / Home couldn't connect to this accessory"

I have tried changing the listen port, the 'username', the pin, removed the 'persist' folder, restarted HomeBridge, but the IOS Home app just won't connect.

I have gone through the similar issues posted here, but don't see an answer for this specific problem ...

**Comment by EnduroFever** [comment]
I eventually got it going without an issue with iOS 10 (10.3 developer beta even).

It did take changing the username out and deleting the persist folder a couple times.  I even rebooted the linux instance and waited for the old advertisements to stop showing up in the Home app.

Eventually I got it connected and it has been solid since paired yesterday.

**Comment by m1ci** [comment]
@talisker-ca same issue here as well. Not sure whats wrong. Maybe the port should be forwarded? are you behind router/firewall? do you forward the port?

**Comment by talisker-ca** [comment]
No, all connections were done behind the firewall, on the same sub-net.  

At the moment I have given up on getting this to work.

 

Cheers,

Paul Neal

**Comment by danieljimenez** [comment]
@EnduroFever is it reliable? In my experience once I do get it to connect, it will stop working within an hour or two.

**Comment by EnduroFever** [comment]
Honestly, it is more reliable than native HomeKit devices for me.  I haven't had a single hiccup since setting it up.

I am using a small dedicated Ubuntu 16.04 LTS instance through VMware on a NUC with an Apple TV for remote access.
